functorch/csrc/BatchingRegistrations.cpp

```diff
@@ -448,40 +448,83 @@ Tensor& zero_inplace_batching_rule(Tensor &self) {
   return self;
 }
 
-Tensor squeeze_batching_rule(const Tensor& self) {
-  if (!participatesInCurrentLevel(self)) {
-    c10::impl::ExcludeDispatchKeyGuard guard(kBatchedKey);
-    return self.squeeze();
-  }
-  auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
-  auto physical_sizes = self_physical.tensor().sizes();
-
-  // Don't squeeze the batch dims!
-  VmapDimVector squeezed_sizes;
-  int64_t num_batch_dims = self_physical.numBatchDims();
-  squeezed_sizes.insert(
-      squeezed_sizes.end(),
-      physical_sizes.begin(),
-      physical_sizes.begin() + num_batch_dims);
-  for (auto it = physical_sizes.begin() + num_batch_dims; it != physical_sizes.end(); ++it) {
-    if (*it != 1) {
-      squeezed_sizes.push_back(*it);
+std::tuple<Tensor, optional<int64_t>> squeeze_batching_rule(const Tensor& self, optional<int64_t> bdim) {
+  if (bdim) {
+    if (self.dim() == 1) {
+      // Special case for scalar arrays to replicate PyTorch behavior.
+      return std::make_tuple<Tensor, optional<int64_t>>(
+        std::move(self.alias()),
+        std::move(bdim)
```

Comment:
Do we need std::move for these? (I'm not very good with C++ so I don't know and the rest of this repo has not been using std::move this way)